gems/pending/VMwareWebService/VimService.rb

```diff
@@ -892,8 +902,19 @@ def retrievePropertiesEx(propCol, specSet, max_objects = nil)
   end
 
   def retrievePropertiesCompat(propCol, specSet, max_objects = nil)
+    oc = VimArray.new('ArrayOfObjectContent')
+
     rv = retrievePropertiesEx(propCol, specSet, max_objects)
-    rv ? rv['objects'] : []
+    if rv
+      oc.concat(rv['objects'])
+
+      while rv && rv['token']
+        rv = continueRetrievePropertiesEx(propCol, rv['token'])
+        oc.concat(rv['objects']) unless rv.nil?
```

Comment:
@agrare - Ah, I confused the two. Given `retrieveProperties` is usually a specific request for some subset of the inventory data, this is probably ok the way it is. The only time I think it may be a problem is when we clear the broker's cache. I believe that will cause the inventory to be re-read via `retrieveProperties` the next time a request comes in.